internal/webserver/type/engine.go

```diff
@@ -0,0 +1,99 @@
+package webserver
+
+import (
+	"context"
+	"fmt"
+
+	webscan "github.com/Method-Security/webscan/generated/go"
+	apacheEnumerationModules "github.com/Method-Security/webscan/internal/webserver/type/apache/enumeration"
+	apacheValidationModules "github.com/Method-Security/webscan/internal/webserver/type/apache/validation"
+	nginxEnumerationModules "github.com/Method-Security/webscan/internal/webserver/type/nginx/enumeration"
+	nginxValidationModules "github.com/Method-Security/webscan/internal/webserver/type/nginx/validation"
+)
+
+type ModuleLibrary interface {
+	ModuleRun(target string, config *webscan.WebServerTypeConfig) (*webscan.Attempt, []string)
+	AnalyzeResponse(response *webscan.ResponseUnion) bool
+}
+
+type TypeEngine struct {
+	Library ModuleLibrary
```

Comment:
Why would we not have the engine hold a list of the Modules?

Reply:
Especially since the list of modules (as written today) will be the same for every single target

Reply:
I would also store the config on the Engine

Reply:
I like the callout, Ill move the logic from returnModules to the engine struct